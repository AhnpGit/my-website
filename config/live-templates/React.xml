<templateSet group="React">
  <template name="rx_module" value="const _ACTION_PREFIX = '$projectName$/$moduleName$'&#10;&#10;const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;&#10;// TODO: Connect the reducer.&#10;export default function reducer(state = {$defaultState$}, action) {&#10;    switch (action.type) {&#10;        case $ACTION_NAME$:&#10;            return {$END$}&#10;&#10;        default:&#10;            return state&#10;    }&#10;}&#10;&#10;export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$payload$},&#10;    meta: {},&#10;    error: undefined,&#10;})&#10;" description="Creates a DUCKS-module" toReformat="true" toShortenFQNames="true">
    <variable name="projectName" expression="" defaultValue="&quot;&lt;project name&gt;&quot;" alwaysStopAt="true" />
    <variable name="moduleName" expression="fileNameWithoutExtension()" defaultValue="" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <variable name="defaultState" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="payload" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component" value="import React from 'react'&#10;&#10;const $ComponentName$ = ({$props$}) =&gt; (&#10;    &lt;$END$/&gt;&#10;)&#10;&#10;export default $ComponentName$" description="Creates a functional component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="props" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component_class" value="import React from 'react'&#10;&#10;class $ComponentName$ extends React.Component {&#10;    render() {&#10;        return (&#10;            &lt;$END$/&gt;&#10;        )&#10;    }&#10;}&#10;&#10;export default $ComponentName$" description="Creates a class component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_proptypes" value="import PropTypes from 'prop-types'&#10;$ComponentName$.propTypes = {&#10;    $prop$: PropTypes.$END$,&#10;}" description="Adds proptypes to a component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="prop" expression="" defaultValue="&quot;prop&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action_async" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;const $ACTION_NAME$_PENDING = `${$ACTION_NAME$}_PENDING`&#10;const $ACTION_NAME$_FULFILLED = `${$ACTION_NAME$}_FULFILLED`&#10;const $ACTION_NAME$_REJECTED = `${$ACTION_NAME$}_REJECTED`" description="Creates async action types" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`" description="Creates an action type" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$END$},&#10;    meta: {},&#10;    error: undefined,&#10;})" description="Creates an action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator_async" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: getPayload(),&#10;    meta: {},&#10;    error: undefined,&#10;    &#10;    async function getPayload() {&#10;        return $END$&#10;    }&#10;})" description="Creates an async action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_container" value="import { connect } from 'react-redux'&#10;import $ComponentName$ from '../components/$ComponentName$'&#10;&#10;const mapStateToProps = (state) =&gt; ({&#10;    $END$&#10;})&#10;&#10;const mapDispatchToProps = (dispatch) =&gt; ({&#10;})&#10;&#10;export default connect(&#10;    mapStateToProps,&#10;    mapDispatchToProps,&#10;)($ComponentName$)" description="Creates a container class" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_styled" value="import styled from 'styled-components'&#10;&#10;const $ComponentName$ = styled.$ElementType$`&#10;  $END$&#10;`&#10;&#10;export default $ComponentName$" description="Creates a styled component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="ElementType" expression="" defaultValue="&quot;div&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>
