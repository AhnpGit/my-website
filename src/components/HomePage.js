import React from 'react'

const HomePage = () => (
  <div>
    <Nav />

  </div>
)

export default HomePage
