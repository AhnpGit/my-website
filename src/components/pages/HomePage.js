// @flow

import React from 'react'

import Cities from '../Cities'
import ContactForm from '../ContactForm'
import Features from '../../containers/Features'
import Hero from '../molecules/Hero/Hero'
import Meals from '../Meals'
import Plans from '../Plans'
import Steps from '../Steps'
import Testimonials from '../Testimonials'

/**
 * Homepage component displays home page
 */
const HomePage = () => (
  <div>
    <Hero />
    <Features />
  </div>
)

export default HomePage
