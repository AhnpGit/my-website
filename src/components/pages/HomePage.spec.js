import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import HomePage from './HomePage'

describe('HomePage component', () => {
  it('renders home page without any issues', () => {
    const component = shallow(
      <HomePage />
    )

    const result = shallowToJson(component)

    expect(result).toMatchSnapshot()
  })
})
