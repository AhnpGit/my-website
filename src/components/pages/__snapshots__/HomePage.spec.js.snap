// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders home page without any issues 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Hero />
    <Connect(Features) />
</div>,
  "nodes": Array [
    <div>
      <Hero />
      <Connect(Features) />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <HomePage />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Hero />
          <Connect(Features) />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Hero />
          <Connect(Features) />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <HomePage />,
}
`;
